File: ios/Classes/ContentShareObserver.swift
import AmazonChimeSDK

class ChimeContentShareObserver: ContentShareObserver {
    let chimeFlutterApi: ChimeFlutterApi

    init(chimeFlutterApi: ChimeFlutterApi) {
        self.chimeFlutterApi = chimeFlutterApi
    }

    func contentShareDidStart() {
        chimeFlutterApi.onContentShareStateChanged(state: 0) { _ in }
    }

    func contentShareDidStop(status: ContentShareStatus) {
        chimeFlutterApi.onContentShareStateChanged(state: 1) { _ in }
    }
}

File: android/src/main/kotlin/com/retozu/flutter_amazon_chime/ContentShareObserver.kt
package com.retozu.flutter_amazon_chime

import com.amazonaws.services.chime.sdk.meetings.audiovideo.contentshare.ContentShareObserver
import com.amazonaws.services.chime.sdk.meetings.audiovideo.contentshare.ContentShareStatus

class ChimeContentShareObserver(private val chimeFlutterApi: ChimeFlutterApi) : ContentShareObserver {
    override fun onContentShareStarted() {
        chimeFlutterApi.onContentShareStateChanged(0L) {}
    }

    override fun onContentShareStopped(status: ContentShareStatus) {
        chimeFlutterApi.onContentShareStateChanged(1L) {}
    }
}
